### DISCUSSION TITLE
Text Normalization for non-En languages

### DISCUSSION THREAD

**Discussion by cahya-wirawan** [NVIDIA-NeMo/NeMo#2260] [Q&A] [answered]
Hi, how can I create a new nemo text normalization for other language? Is there a tutorial to do it? Thanks

**Comment by ekmb** [comment, marked as answer]
Hi @cahya-wirawan, there is no tutorial on how to create text normalization for a new language yet. This is something we plan to add in future releases. What language(s) are you interested in?

**Reply by cahya-wirawan** [reply]
Hi @ekmb, I would like to have the text normalization for Indonesian and German. If there is an example to do it for another language, I could contribute for Indonesian.

**Reply by yzhang123** [reply]
hi @cahya-wirawan we have Gemanfor both itn and tn! https://github.com/NVIDIA/NeMo/tree/main/nemo_text_processing/text_normalization/de

Here is a tutorial on how to create your own grammar and we have a Vietnamese ITN open source contribution already. So you are more than welcome to contribute Indonesian, we would highly appreciate it. Here is a very detailed tutorial on how to do this  https://github.com/NVIDIA/NeMo/blob/main/tutorials/text_processing/WFST_Tutorial.ipynb

Please ping me if you have questions.

**Reply by Approximetal** [reply]
> hi @cahya-wirawan we have Gemanfor both itn and tn! https://github.com/NVIDIA/NeMo/tree/main/nemo_text_processing/text_normalization/de
> 
> Here is a tutorial on how to create your own grammar and we have a Vietnamese ITN open source contribution already. So you are more than welcome to contribute Indonesian, we would highly appreciate it. Here is a very detailed tutorial on how to do this https://github.com/NVIDIA/NeMo/blob/main/tutorials/text_processing/WFST_Tutorial.ipynb
> 
> Please ping me if you have questions.

Hi, these pages cannot open, could you update the url please?

**Comment by ekmb** [comment]
@Approximetal , TN/ITN was moved to a separate repo https://github.com/NVIDIA/NeMo-text-processing
tutorials: https://github.com/NVIDIA/NeMo-text-processing/tree/main/tutorials
docs: https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/nlp/text_normalization/intro.html
